diff -Naur openssl-1.0.1e/crypto/evp/evp_test.c openssl-1.0.1e-mod/crypto/evp/evp_test.c
--- openssl-1.0.1e/crypto/evp/evp_test.c	2013-02-11 16:26:04.000000000 +0100
+++ openssl-1.0.1e-mod/crypto/evp/evp_test.c	2013-10-18 13:31:25.469714257 +0200
@@ -347,8 +347,8 @@
     /* Load all compiled-in ENGINEs */
     ENGINE_load_builtin_engines();
 #endif
-#if 0
-    OPENSSL_config();
+#if 1
+    OPENSSL_config(NULL);
 #endif
 #ifndef OPENSSL_NO_ENGINE
     /* Register all available ENGINE implementations of ciphers and digests.
