From 2641ee5520a483af9ac481365dea679f51f2a020 Mon Sep 17 00:00:00 2001
From: Karl Hiramoto <karl@hiramoto.org>
Date: Fri, 24 Jan 2014 17:26:14 +0100
Subject: [PATCH 1/4] Exokey has 256M of ram

---
 include/configs/sama5d3xek.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/sama5d3xek.h b/include/configs/sama5d3xek.h
index d22b0e7..dbe7d55 100644
--- a/include/configs/sama5d3xek.h
+++ b/include/configs/sama5d3xek.h
@@ -122,7 +122,7 @@
 /* SDRAM */
 #define CONFIG_NR_DRAM_BANKS		1
 #define CONFIG_SYS_SDRAM_BASE           ATMEL_BASE_DDRCS
-#define CONFIG_SYS_SDRAM_SIZE		0x20000000
+#define CONFIG_SYS_SDRAM_SIZE		0x10000000
 
 #define CONFIG_SYS_INIT_SP_ADDR \
 	(CONFIG_SYS_SDRAM_BASE + 4 * 1024 - GENERATED_GBL_DATA_SIZE)
-- 
1.8.3.2

