diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index a28f6ff..5227c04 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -147,8 +147,10 @@ static inline bool has_rndis(void)
  * used with CDC Ethernet, Linux 2.4 hosts will need updates to choose
  * the non-RNDIS configuration.
  */
-#define RNDIS_VENDOR_NUM	0x0525	/* NetChip */
-#define RNDIS_PRODUCT_NUM	0xa4a2	/* Ethernet/RNDIS Gadget */
+
+#define RNDIS_VENDOR_NUM	0x29B7	/* XoWare Inc */
+#define RNDIS_PRODUCT_NUM	0x0101	/* Ethernet/RNDIS Gadget */
+
 
 /* For EEM gadgets */
 #define EEM_VENDOR_NUM		0x1d6b	/* Linux Foundation */
